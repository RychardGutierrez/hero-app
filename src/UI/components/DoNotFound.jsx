import { useNavigate } from 'react-router-dom';

import { handleReturnHome } from './../tools/handles';

const DoNotFound = () => {
  const navigate = useNavigate();

  return (
    <>
      <h1 className="text-center">Ups Hero Do not found</h1>
      <div className="text-center">
        <button
          type="button"
          className="btn btn-link"
          onClick={() => handleReturnHome(navigate)}
        >
          Return Home
        </button>
      </div>
    </>
  );
};

export default DoNotFound;
