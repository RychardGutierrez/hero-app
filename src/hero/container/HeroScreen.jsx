import { useMemo } from 'react';
import { useParams, Navigate, useNavigate } from 'react-router-dom';
import { getImageHero } from '../../shared/constants';
import { handleReturnNavigate } from '../../UI/tools/handles';
import { getHeroById } from './../../shared/selectors';

const HeroScreen = () => {
  const navigate = useNavigate();

  const { idHero } = useParams();
  const hero = useMemo(() => getHeroById(idHero), [idHero]);

  if (!hero) {
    return <Navigate to="/404" />;
  }

  const { alter_ego, characters, first_appearance, id, publisher, superhero } =
    hero;

  return (
    <div className="row mt-5">
      <div className="col-4">
        <img
          src={getImageHero(id)}
          alt={superhero}
          className="img-thumbnail animate__animated animate__fadeInLeft"
        />
      </div>
      <div className="col-8">
        <h3>{superhero}</h3>
        <ul className="list-group list-group-flush">
          <li className="list-group-item">
            <b>Alter ego:</b> {alter_ego}
          </li>
          <li className="list-group-item">
            <b>Publisher:</b> {publisher}
          </li>
          <li className="list-group-item">
            <b>First Appearance:</b> {first_appearance}
          </li>
        </ul>
        <h5 className="mt-3"> Characteres</h5>
        <p>{characters}</p>
        <button
          className="btn btn-outline-info"
          onClick={() => handleReturnNavigate(navigate)}
        >
          Return
        </button>
      </div>
    </div>
  );
};

export default HeroScreen;
