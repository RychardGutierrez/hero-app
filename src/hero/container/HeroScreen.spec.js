import { mount } from 'enzyme';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import DoNotFound from '../../UI/components/DoNotFound';
import HeroScreen from './HeroScreen';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('HeroScreen', () => {
  it('should render component', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/hero']}>
        <Routes>
          <Route path="/hero" element={<HeroScreen />} />
          <Route path="/404" element={<DoNotFound />} />
        </Routes>
      </MemoryRouter>,
    );

    expect(wrapper.find('h1').text().trim()).toBe('Ups Hero Do not found');

    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with hero', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/hero/marvel-spider']}>
        <Routes>
          <Route path="/hero/:idHero" element={<HeroScreen />} />
          <Route path="/404" element={<DoNotFound />} />
        </Routes>
      </MemoryRouter>,
    );

    expect(wrapper.find('h3').text().trim()).toBe('Spider Man');

    expect(wrapper).toMatchSnapshot();
  });

  it('should return the previus view', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/hero/marvel-spider']}>
        <Routes>
          <Route path="/hero/:idHero" element={<HeroScreen />} />
        </Routes>
      </MemoryRouter>,
    );
    wrapper.find('button').prop('onClick')();
    expect(mockNavigate).toBeCalledWith(-1);
  });

  it('should return message the hero do not found', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/hero/hero-do-not-exist']}>
        <Routes>
          <Route path="/hero/:idHero" element={<HeroScreen />} />
          <Route path="/404" element={<DoNotFound />} />
        </Routes>
      </MemoryRouter>,
    );

    expect(wrapper.find('h1').text().trim()).toBe('Ups Hero Do not found');
  });
});
