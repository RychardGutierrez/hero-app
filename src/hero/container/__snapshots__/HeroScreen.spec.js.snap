// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroScreen should render component 1`] = `
Array [
  <h1
    className="text-center"
  >
    Ups Hero Do not found
  </h1>,
  <div
    className="text-center"
  >
    <button
      className="btn btn-link"
      onClick={[Function]}
      type="button"
    >
      Return Home
    </button>
  </div>,
]
`;

exports[`HeroScreen should render component with hero 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-4"
  >
    <img
      alt="Spider Man"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src="/assets/marvel-spider.jpg"
    />
  </div>
  <div
    className="col-8"
  >
    <h3>
      Spider Man
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <b>
          Alter ego:
        </b>
         
        Peter Parker
      </li>
      <li
        className="list-group-item"
      >
        <b>
          Publisher:
        </b>
         
        Marvel Comics
      </li>
      <li
        className="list-group-item"
      >
        <b>
          First Appearance:
        </b>
         
        Amazing Fantasy #15
      </li>
    </ul>
    <h5
      className="mt-3"
    >
       Characteres
    </h5>
    <p>
      Peter Parker
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
</div>
`;
